connector.go

```diff
@@ -234,6 +243,10 @@ func (c *Connector) ProduceAndCommitMessage(namespace, subject string, msgType k
 func (c *Connector) startProducer() error {
 	txID := c.id()
 
+	if c.isBackfill {
+		txID = fmt.Sprintf("%s%d", txID, rand.Int())
```

Comment:
This doesn't ensure uniqueness. You will need to use a uuid instead.

Reply:
If you still want to use rand then you need to call rand.Seed() first or it will always generate the same number for every connector.

Reply:
I think you mean same sequence of numbers, not the exact same number.  I am surprised the implementation of Golangs' rand() function is so primitive if it will generate the same number or even the same sequence.  Most modern random number generators self seed with some entropy from the microprocessor they are running on.

Reply:
I remembered later that math/rand lib has a lot of issues. crypto/rand lib delivers what's expected from a randomizer.